src/main/java/org/apache/hadoop/hive/ql/io/parquet/timestamp/NanoTimeUtils.java

```diff
@@ -10,13 +11,15 @@
     private NanoTimeUtils() {}
 
     private static final int JULIAN_EPOCH_OFFSET_DAYS = 2_440_588;
+    private static final long SECONDS_PER_DAY = 86400L;
 
     public static NanoTime getNanoTime(Timestamp timestamp, @SuppressWarnings("unused") boolean ignored)
     {
-        int epochDay = toIntExact(SECONDS.toDays(timestamp.toEpochSecond()));
+        long epochSeconds = timestamp.toEpochSecond();
+        int epochDay = toIntExact(floorDiv(epochSeconds, SECONDS_PER_DAY));
         int julianDay = JULIAN_EPOCH_OFFSET_DAYS + epochDay;
 
-        long timeOfDaySeconds = timestamp.toEpochSecond() % 86400;
+        long timeOfDaySeconds = epochSeconds - epochDay * SECONDS_PER_DAY;
```

Comment:
This would be simpler as
```java
long timeOfDaySeconds = floorMod(epochSeconds, SECONDS_PER_DAY);

Reply:
I used that initially, but Intellij highlighted it as not being available in Java 8.  It turns out that was the case only for the overload taking `long` and `int` as arguments, but I didn't realize, so I rolled my own...

Reply:
Let's switch to that. The `(long, long)` version is fine (since the result is `long`)

Reply:
Done.